prov/shm/src/smr_progress.c

```diff
@@ -319,7 +319,7 @@ static int smr_mmap_peer_copy(struct smr_ep *ep, struct smr_cmd *cmd,
 
 	if (cmd->msg.hdr.op == ofi_op_read_req) {
 		*total_len = ofi_total_iov_len(iov, iov_count);
-		if (ofi_copy_from_iov(mapped_ptr, *total_len, iov, iov_count, 0)
+		if (ofi_copy_from_iov(mapped_ptr, cmd->msg.hdr.size, iov, iov_count, 0)
 		    != *total_len) {
```

Comment:
Why calculate total_len here?  The same logic as the 'else' branch can be used here.

Reply:
I didn't look too hard into why that is being done, but my understanding based on a quick scan was that `total_len` here stores the total number of bytes that should be read into `mapped_ptr`. I assume that value was not readily available elsewhere if it is being calculated here. It seems correct to me to check that the number of bytes copied is the number of bytes expected to be read, and error with `-FI_ETRUNC` if that check fails.

Reply:
my understanding is cmd->msg.hdr.size is the size to read and it's fine there are more data at the source then needed. Not sure why the original code use the source buffer size instead. @aingerson could you check?

Reply:
You are both correct! cmd->msg.hdr.size should be used and total_len shouldn't need to be calculated

Reply:
:) I will fix that up